### DISCUSSION TITLE
Reading from I2C sensor and generating DigitalOutput with same Arduino

### DISCUSSION THREAD

**Discussion by tomasmanu** [bonsai-rx/bonsai#1346] [Q&A]
Hi there!
  Currently I'm using two Arduinos in my setup: one receives data from a sensor through I2C protocol and prints it to Serial (I then read this data with Bonsai using `CreateSerialPort` and `SerialStringRead`); the other one controls a LED through StandardFirmata protocol with Bonsai (using `DigitalOutput`).
 My question is if these two tasks can be accomplished using only ONE Arduino. What is the best way to implement this?  I don't know how to read from an I2C sensor through StandardFirmata protocol. Because of that I decided to use two different boards.
 I leave attached the workflow I'm using for the two Arduinos' approach.
Many thanks,
Tomás 
[read_from_arduino_serial_and_digital_output_to_another_arduino.zip](https://github.com/bonsai-rx/bonsai/files/11392563/read_from_arduino_serial_and_digital_output_to_another_arduino.zip)

**Comment by bruno-f-cruz** [comment]
You can implement the I2C communication in the firmata code, or implement the digital control logic in your current Arduino I2C sensor sketch. The easiest is probably the latter. Use SerialStringWrite to send a specific string to turn on the LED and a different one to turn it off.

**Reply by tomasmanu** [reply]
Hi Bruno
 I tried to implement the latter approach, but I'm having problems to turn off the LED with the other string. I leave attached both the Arduino sketch and the Bonsai workflow. 
In order to make the sketch simpler, I changed the `Serial.print` of the I2C sensor data with `Serial.print("hello")`.
In the Bonsai workflow I generate two strings: 'o' for turning on, and 'f' for turning off the LED on pin 13 of the board. The LED is turned on when I press the corresponding key (E). However, when I press the key to turn it off (A), the Arduino board registers the `Serial.write` event (as the RX led blinks once), but it doesn´t turn off the LED on pin 13. I have to press both keys (first E and then A) in order to turn off the LED. What am I doing wrong?
Thanks
[read_write_serial.zip](https://github.com/bonsai-rx/bonsai/files/11403066/read_write_serial.zip)
